src/merkle/smt/full/concurrent/mod.rs

```diff
@@ -357,11 +367,10 @@ impl Smt {
     fn process_sorted_pairs_to_leaves<F>(
         pairs: Vec<(RpoDigest, Word)>,
         mut process_leaf: F,
-    ) -> PairComputations<u64, SmtLeaf>
+    ) -> Result<PairComputations<u64, SmtLeaf>, MerkleError>
```

Comment:
We should update the doc comments for this function to explain under what conditions an error could be returned.